src/test/java/seedu/address/logic/commands/ReassignCommandTest.java

```diff
@@ -0,0 +1,201 @@
+        LocalDate startDate = TODAY_TOMORROW.getStartDate();
+        RoomNumber newRoomNumber = ROOM_NUMBER_031;
+        ReassignCommand reassignCommand = new ReassignCommand(roomNumber, startDate, newRoomNumber);
+
+        Model expectedModel = new ModelManager(model.getConcierge(), new UserPrefs());
+        expectedModel.reassignRoom(roomNumber, startDate, newRoomNumber);
+        expectedModel.commitConcierge();
+
+        // reassign -> booking reassigned
+        reassignCommand.execute(model, commandHistory);
+
+        // undo -> reverts concierge back to previous state
+        expectedModel.undoConcierge();
+        assertCommandSuccess(new UndoCommand(), model, commandHistory, UndoCommand.MESSAGE_SUCCESS, expectedModel);
+
+        // redo -> same booking reassigned
+        expectedModel.redoConcierge();
+        assertCommandSuccess(new RedoCommand(), model, commandHistory, RedoCommand.MESSAGE_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void executeUndoRedo_invalidReassign_failure() {
+        RoomNumber roomNumber = ROOM_NUMBER_011;
+        LocalDate startDate = TODAY_TOMORROW.getStartDate();
+        RoomNumber newRoomNumber = ROOM_NUMBER_020;
+        ReassignCommand reassignCommand = new ReassignCommand(roomNumber, startDate, newRoomNumber);
+
+        String expectedMessage = String.format(ReassignCommand.MESSAGE_OVERLAPPING_BOOKING, newRoomNumber);
+
+        assertCommandFailure(reassignCommand, model, commandHistory, expectedMessage);
+
+        // single Concierge state in model -> undoCommand and redoCommand fail
+        assertCommandFailure(new UndoCommand(), model, commandHistory, UndoCommand.MESSAGE_FAILURE);
+        assertCommandFailure(new RedoCommand(), model, commandHistory, RedoCommand.MESSAGE_FAILURE);
+    }
+
+    @Test
+    public void equals() {
+        RoomNumber firstRoomNumber = ROOM_NUMBER_010;
+        RoomNumber secondRoomNumber = ROOM_NUMBER_011;
+        LocalDate validStartDate = TODAY_TOMORROW.getStartDate();
+        RoomNumber validRoomNumberToReassignTo = ROOM_NUMBER_031;
+
+        ReassignCommand reassignFirstCommand = new ReassignCommand(firstRoomNumber, validStartDate,
+                validRoomNumberToReassignTo);
+        ReassignCommand reassignSecondCommand = new ReassignCommand(secondRoomNumber, validStartDate,
+                validRoomNumberToReassignTo);
+
+        // same object -> returns true
+        assertTrue(reassignFirstCommand.equals(reassignFirstCommand));
+
+        // same values -> returns true
+        ReassignCommand reassignFirstCommandCopy = new ReassignCommand(firstRoomNumber, validStartDate,
+                validRoomNumberToReassignTo);
+        assertTrue(reassignFirstCommand.equals(reassignFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(reassignFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(reassignFirstCommand.equals(null));
```

Comment:
assertTrue(reassignFirstCommand != null)

Reply:
actually, no. this test is testing the `equals` method, so it needs to be used like this. you can refer to all other equals test in other classes